### ISSUE TITLE
[Vulnerability] The pbsrun_wrap/pbsrun_unwrap may cause vulnerability issues.

### ISSUE THREAD

**Issue by zhenrong-wang** [openpbs/openpbs#2637]
The pbsrun_wrap/pbsrun_unwrap scripts may cause arbitrary move of system binaries:

**Code block:**

In /src/cmds/pbsrun_wrap.in :

```
if ! [ -x $actual_mpirun ] ; then
   echo "$progname: $actual_mpirun not exists and an executable!"
   exit 1
fi

actual_mpirun_name=`basename $actual_mpirun`
actual_mpirun_dir=`dirname $actual_mpirun`

# Save original mpirun script
res=`exec_cmd mv $actual_mpirun ${actual_mpirun}.actual`
```

**Analysis**

The script only checks whether the executable provided by user/admin is executable or not, which means that any executable files could be accepted and **MOVED**. This may cause severe risks when a user/admin puts an important system utilitie as the argument $2. 

E.g.

`sudo ./pbsrun_wrap /usr/bin/rm pbsrun.poe`

This would move the system utility (GNU coreutils - rm) to /usr/bin/rm.actual

**Suggestion**

The script should check the provided executable in a stricter way. I'd suggesst to add the scripts below:

```
# Make sure the provided $actual_mpirun is a valid mpirun

res1=`$actual_mpirun --help | grep -w mpirun`
res2=`$actual_mpirun --help | grep -w ./mpiexec`
if [ -z '$res1' ] && [ -z '$res2' ]; then
   echo "$progname: $actual_mpirun is not a valid mpirun executable!"
   exit 1
fi

```

However, this method needs full tests because different mpirun distributions print help info in different ways. OpenMPI and MPICH have been validated by me, but I am not sure whether it is applicable for other mpirun distributions.

Anyway, I'd suggest this issue should be considered and addressed.

Zhenrong WANG

**Comment by sourceonly** [comment]
Yeah, i agree you could polish it, but i doubt it is necessary.  Wrap the 'rm' executable makes no sense if you knew it's not a mpi executable.
